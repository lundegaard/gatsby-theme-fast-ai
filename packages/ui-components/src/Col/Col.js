import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { isNil } from 'ramda';

import mapResponsiveProperty from '../utils/mapResponsiveProperty';
import * as Types from '../types';
import Box from '../Box';
import useTheme from '../hooks/useTheme';

const spanToWidth = (maxColumns) => (columnSpan) => columnSpan / maxColumns;

const Col = forwardRef(({ span, maxColumns: maxColumnsProp, ...rest }, ref) => {
	const { grid: { gutters, maxColumns } = {} } = useTheme();

	return (
		<Box
			ref={ref}
			px={gutters}
			width={mapResponsiveProperty(
				spanToWidth(isNil(maxColumnsProp) ? maxColumns : maxColumnsProp),
				span
			)}
			{...rest}
		/>
	);
});
Col.displayName = 'Col';

Col.propTypes = {
	maxColumns: PropTypes.number,
	span: Types.responsiveNumber,
};

export default Col;
