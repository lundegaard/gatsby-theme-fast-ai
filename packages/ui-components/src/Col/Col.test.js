import React from 'react';
import { shallow } from 'enzyme';

import Col from './Col';

describe('<Col />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Col {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
