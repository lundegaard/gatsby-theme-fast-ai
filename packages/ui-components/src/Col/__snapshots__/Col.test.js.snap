// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> matches snapshot when shallowly rendered 1`] = `
<Box
  px={2}
  width={
    Array [
      NaN,
    ]
  }
/>
`;
